daemon_test.go

```diff
@@ -0,0 +1,211 @@
+	{500, []byte(`{
+"error": "Invalid request"
+}`), false, false, nil},
+	{200, []byte(malformedUpdateResponse), false, false, nil},
+	{200, []byte(brokenUpdateResponse), false, false, nil},
+	{200, []byte(missingFieldsUpdateResponse), false, false, nil},
+}
+
+func TestParseUpdateResponse(t *testing.T) {
+	updateRequester := updateRequester{
+		reqType:              http.MethodGet,
+		request:              "",
+		menderClient:         client{},
+		updateResponseParser: parseUpdateResponse,
+	}
+
+	for _, tt := range updateTest {
+		uAPIResp, err := updateRequester.updateResponseParser(http.Response{StatusCode: tt.responseStatusCode}, []byte(tt.responseBody))
+		if tt.shoulReturnError && err != nil {
+			t.Fatal("Update parsing should not return error but it does: ", err)
+		} else if !tt.shoulReturnError && err == nil {
+			t.Fatal("Update parsing should return an error but is not.")
+		}
+		if tt.shouldCheckReturnType && reflect.TypeOf(uAPIResp) != reflect.TypeOf(tt.returnType) {
+			t.Fatal("Update parse returned unexpected type: ", reflect.TypeOf(uAPIResp), " expecting: ", reflect.TypeOf(tt.returnType))
+		}
+	}
+}
+
+type fakeUpdateType int
+
+func (updateAPIResp fakeUpdateType) actOnData() error {
+	return nil
+}
+
+func fakeParseUpdateResponse(response http.Response, respBody []byte) (dataActuator, error) {
+	var fakeData fakeUpdateType
+	return fakeData, nil
+}
+
+func TestGetUpdate(t *testing.T) {
+	// Test server that always responds with 200 code, and specific payload
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Header().Set("Content-Type", "application/json")
+		//TODO
+		fmt.Fprint(w, correctUpdateResponse)
+	}))
+	defer ts.Close()
+
+	client, _ := initClient(authCmdLineArgsType{ts.URL, "client.crt", "client.key", "server.crt"})
+	var config daemonConfigType
+	config.deviceID = "fake_id"
+
+	// test code with real update requester
+	updateRequester := updateRequester{
+		reqType:              http.MethodGet,
+		request:              ts.URL + "/" + config.deviceID + "/update",
+		menderClient:         client,
+		updateResponseParser: fakeParseUpdateResponse,
+	}
+
+	if err := makeJobDone(updateRequester); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCheckPeriodicDaemonUpdate(t *testing.T) {
+
+	reqHandlingCnt := 0
+	pullInterval := time.Duration(100) * time.Millisecond
+
+	// Test server that always responds with 200 code, and specific payload
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintln(w, correctUpdateResponse)
+		reqHandlingCnt++
+	}))
+	defer ts.Close()
+
+	client, _ := initClient(authCmdLineArgsType{ts.URL, "client.crt", "client.key", "server.crt"})
+	var config daemonConfigType
+	config.serverPullInterval = pullInterval
+	config.server = ts.URL
+	config.deviceID = "fake_id"
+
+	go func() {
```

Comment:
you'd use 

```
z := "Time is"
go func() {
   fmt.Println(z, time.Now())
}()
```

if you want to put inline code that will be the routine or want to use closure